### PR TITLE
S3 repo and shard allocation awareness

### PR DESCRIPTION
**Pull request by tarvip**
Added plugin for sending snapshots to s3
Also parameters for shard allocation awareness across availability zones (more info can be seen here https://www.elastic.co/guide/en/elasticsearch/reference/current/allocation-awareness.html)

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+# Determine zone name
+KUBE_TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+KUBE_NODE=$(curl -s --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H "Authorization: Bearer $KUBE_TOKEN" https://kubernetes.default/api/v1/namespaces/$NAMESPACE/pods/$HOSTNAME | jq -r '.spec.nodeName')
+ZONE=$(curl -s --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H "Authorization: Bearer $KUBE_TOKEN" https://kubernetes.default/api/v1/nodes/$KUBE_NODE | jq -r '.metadata.labels."failure-domain.beta.kubernetes.io/zone"')
+export ZONE
+
+# provision elasticsearch user
+addgroup sudo
```
Comment (willmore) [inline run.sh]: Why is user setup here instead of in the Dockerfile?

Comment (tarvip) [inline run.sh]: No idea, I took `run.sh` from original repo (pires/docker-elasticsearch) and added my own modifications.

Comment (willmore) [inline run.sh]: Any reason we shouldn't added it to the Dockerfile? This is essentially static image config - seems like Dockerfile is the "right" place to put it.

Comment (tarvip) [inline run.sh]: I agree, I guess also pod startup is slightly faster. Updated.

File: Dockerfile [hunk 2]
```diff
@@ -1,16 +1,30 @@
 FROM quay.io/pires/docker-elasticsearch:5.1.1_1
 
-MAINTAINER pjpires@gmail.com
+MAINTAINER techmovers@salemove.com
+
+# Add user/group
+RUN addgroup sudo
```
Comment (willmore) [inline Dockerfile]: All the `RUN` directive could be joined. I'll do that in a separate PR.

**Pull request merged**